packages/office-ui-fabric-react/src/components/ColorPicker/ColorSlider/ColorSlider.base.tsx

```diff
@@ -51,22 +52,32 @@ export class ColorSlider extends BaseComponent<IColorSliderProps, IColorSliderSt
   }
 
   public render(): JSX.Element {
-    const { className, minValue, maxValue, overlayStyle } = this.props;
-    const { currentValue, isAdjusting } = this.state;
+    const { isAlpha, minValue, maxValue, overlayStyle, theme, className, styles } = this.props;
+    const { currentValue } = this.state;
+
+    const classNames = getClassNames(styles!, {
+      theme: theme!,
+      className
+    });
 
     const currentPercentage = (100 * (currentValue! - minValue!)) / (maxValue! - minValue!);
 
+    const hueStyle = {
+      background:
+        'linear-gradient(to left,red 0,#f09 10%,#cd00ff 20%,#3200ff 30%,#06f 40%,#00fffd 50%,#0f6 60%,#35ff00 70%,#cdff00 80%,#f90 90%,red 100%)'
+    };
+
+    const alphaStyle = {
+      backgroundImage:
+        'url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAAAJUlEQVQYV2N89erVfwY0ICYmxoguxjgUFKI7GsTH5m4M3w1ChQC1/Ca8i2n1WgAAAABJRU5ErkJggg==)'
+    };
+
+    const sliderStyle = isAlpha ? alphaStyle : hueStyle;
+
     return (
-      <div
-        ref={this._root}
-        className={css('ms-ColorPicker-slider', styles.slider, className, isAdjusting && 'is-adjusting')}
-        onMouseDown={this._onMouseDown}
-      >
-        <div className={css('ms-ColorPicker-sliderOverlay', styles.sliderOverlay)} style={overlayStyle} />
-        <div
-          className={css('ms-ColorPicker-thumb is-slider', styles.thumb, styles.thumbIsSlider)}
-          style={{ left: currentPercentage + '%' }}
-        />
+      <div ref={this._root} className={classNames.root} onMouseDown={this._onMouseDown} style={sliderStyle}>
+        <div className={classNames.sliderOverlay} style={overlayStyle} />
+        <div className={classNames.sliderThumb} style={{ left: currentPercentage + '%' }} />
```

Comment:
Could these styles along with the currentPercentage calculation be moved to getStyles?

Reply:
yes!